inc/ruleticket.class.php

```diff
@@ -225,16 +225,20 @@ function executeActions($output, $params, array $input = []) {
 
                   // special case of itil solution template
                   if ($action->fields["field"] == 'solution_template') {
-                     //load template
-                     $template = new SolutionTemplate();
-                     if ($template->getFromDB($action->fields["value"]) && $output['id'] != null) {
-                        $solution = new ITILSolution();
-                        $solution->add([
-                           "itemtype" => "Ticket",
-                           "solutiontypes_id" => $template->getField("solutiontypes_id"),
-                           "content" => Toolbox::addslashes_deep($template->getField('content')),
-                           "status" => CommonITILValidation::WAITING,
-                           "items_id" => $output['id']]);
+                     if (isset($this->options['condition']) && $this->options['condition'] === self::ONADD) {
+                        $output['_solutiontemplates_id'] = $action->fields["value"];
+                     } else {
+                        //load template
+                        $template = new SolutionTemplate();
+                        if ($template->getFromDB($action->fields["value"]) && $output['id'] != null) {
+                           $solution = new ITILSolution();
+                           $solution->add([
+                              "itemtype" => "Ticket",
+                              "solutiontypes_id" => $template->getField("solutiontypes_id"),
+                              "content" => Toolbox::addslashes_deep($template->getField('content')),
+                              "status" => CommonITILValidation::WAITING,
+                              "items_id" => $output['id']]);
+                        }
```

Comment:
I am not sure that creatinng a solution here is a good idea. Maybe should should only set `$output['_solutiontemplates_id']` here, whenever you are on ADD or UPDATE case, and create the solution on `post_addItem` and `post_updateItem` methods, like it is done for tasks and followups (at least in master branch).